bluemira/optimisation/_algorithm.py

```diff
@@ -13,22 +13,44 @@
 
 from __future__ import annotations
 
-import enum
 from dataclasses import asdict, dataclass, field
+from enum import Enum, auto
 
 
-class Algorithm(enum.Enum):
+class Algorithm(Enum):
     """Enumeration of available optimisation algorithms."""
 
-    SLSQP = enum.auto()
-    COBYLA = enum.auto()
-    SBPLX = enum.auto()
-    MMA = enum.auto()
-    BFGS = enum.auto()
-    DIRECT = enum.auto()
-    DIRECT_L = enum.auto()
-    CRS = enum.auto()
-    ISRES = enum.auto()
+    SLSQP = auto()
+    COBYLA = auto()
+    SBPLX = auto()
+    MMA = auto()
+    BFGS = auto()
+    DIRECT = auto()
+    DIRECT_L = auto()
+    CRS = auto()
+    ISRES = auto()
+
+    NELDER_MEAD = auto()
+    POWELL = auto()
+    CG = auto()
+    BFGS_SCIPY = auto()
+    NEWTON_CG = auto()
+    L_BFGS_B = auto()
+    TNC = auto()
+    COBYLA_SCIPY = auto()
+    COBYQA = auto()
+    SLSQP_SCIPY = auto()
+    TRUST_CONSTR = auto()
+    DOGLEG = auto()
+    TRUST_NCG = auto()
+    TRUST_EXACT = auto()
+    TRUST_KRYLOV = auto()
+
+    def __new__(cls, value):
+        """Create Enum and debug attribute"""
+        obj = object.__new__(cls)
+        obj._value_ = value
+        return obj
 
```

Comment:
Don't we need to add default conditions later in this file for the new algorithms?